pkg/util/tls/tls.go

```diff
@@ -74,12 +72,27 @@ func (cfg *ClientConfig) GetTLSConfig() (*tls.Config, error) {
 	return config, nil
 }
 
+type ClientConfigWithEnabled struct {
+	ClientConfig `yaml:",inline"`
+	Enabled      bool `yaml:"tls_enabled"`
+}
+
+// RegisterFlagsWithPrefix registers flags with prefix.
+func (cfg *ClientConfigWithEnabled) RegisterFlagsWithPrefix(prefix string, f *flag.FlagSet) {
+	cfg.ClientConfig.RegisterFlagsWithPrefix(prefix, f)
+	f.BoolVar(&cfg.Enabled, prefix+".tls-enabled", false, "Enable TLS in the GRPC client. This flag is needs to be enabled when any other TLS flag is set. This flag will only be respected by the GRPC client, as for the HTTP client the scheme used in the URL will take precedence.")
+}
+
 // GetGRPCDialOptions creates GRPC DialOptions for TLS
-func (cfg *ClientConfig) GetGRPCDialOptions() ([]grpc.DialOption, error) {
-	if tlsConfig, err := cfg.GetTLSConfig(); err != nil {
+func (cfg *ClientConfigWithEnabled) GetGRPCDialOptions() ([]grpc.DialOption, error) {
+	tlsConfig, err := cfg.GetTLSConfig()
```

Comment:
This can return TLS-related error even if TLS is not enabled. I don't think we should let that happen.

Reply:
This should be resolved as of 342e17c592e6c63c3d01227732a7ea0ebdb27111